src/bun.js/api/bun/h2_frame_parser.zig

```diff
@@ -2466,13 +2425,14 @@ pub const H2FrameParser = struct {
             if (this.remoteSettings) |s| s.initialWindowSize else DEFAULT_WINDOW_SIZE,
             this.paddingStrategy,
         );
-        const ctx_value = this.strong_ctx.get() orelse return entry.value_ptr;
-        const callback = this.handlers.onStreamStart;
-        if (callback != .zero) {
-            // we assume that onStreamStart will never mutate the stream hash map
-            _ = callback.call(this.handlers.globalObject, ctx_value, &[_]jsc.JSValue{ ctx_value, jsc.JSValue.jsNumber(streamIdentifier) }) catch |err|
-                this.handlers.globalObject.reportActiveExceptionAsUnhandled(err);
-        }
+        const this_value = this.strong_this.tryGet() orelse return entry.value_ptr;
+        const ctx_value = js.gc.context.get(this_value) orelse return entry.value_ptr;
+        const callback = js.gc.onStreamStart.get(this_value) orelse return entry.value_ptr;
+
```

Comment:
future project: our generator should have a setting for non-optional write barriers. these will never be null if `this_value` exists